### PR TITLE
FOCUS #18: Sub Account Name dimension

### PR DESCRIPTION
**Pull request by MichaelArkoosh-vega**
FOCUS #18: Sub Account Name dimension

### CODE REVIEW

**Review by rileyjenk — CHANGES_REQUESTED** [review]
Sub Accounts can generate invoices.

**Comment by MichaelArkoosh-vega** [comment]
> Sub Accounts can generate invoices.

Where they can generate and invoice (meaning usage in the account can result in an invoice (example I purchase an RI in sub account A.) I think we may need to change the terminology to "Sub accounts must be associated with a billing account as they do not **receive** invoices." - Thoughts?

**Comment by MichaelArkoosh-vega** [comment]
> @udam-f2 They do roll up to the billing account - but that is why I was wondering if we wanted to move from generate to receive? As the underlying usage the invoice is generated from is in the the sub account but the invoice is received at the billing account.

**Comment by udam-f2** [comment]
> > @udam-f2 They do roll up to the billing account - but that is why I was wondering if we wanted to move from generate to receive? As the underlying usage the invoice is generated from is in the the sub account but the invoice is received at the billing account.

Ah, got it now.. then yes, that makes sense. The only odd case we may need to verify is MSP type cases? Seems like they would be fine as well since the invoice is received by the MSP from the billing account. They may generate a different invoice for their underlying customers.

**Comment by MichaelArkoosh-vega** [comment]
Updated the verbiage in the spec document to state "Sub Accounts do not receive an invoice" as, even though the invoice may show in the sub account it is the billing account that will receive/pay said invoice.

File: specification/dimensions/subaccountname.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Spec Content
+------------
+
+### Sub Account Name
+
+A sub account is an optional provider-supported construct for organizing resources and services connected to a billing account. Some common scenarios for using sub accounts include grouping based on organizational constructs and access management boundaries. Sub accounts must be associated with a billing account as they do not receive invoices.
+
+A sub account name is a display name assigned to a sub account. It is commonly used for cost allocation, reporting, and chargeback purposes.
+
+The SubAccountName column MUST be present in the billing data. This column MUST be of type String and MUST NOT contain null values. Where a provider doesn't support a sub account construct (only has a billing account), use the BillingAccountName value.
```
Comment (flanakin) [inline specification/dimensions/subaccountname.md:14]: Does this convey what we discussed?
```suggestion
The SubAccountName column MUST be present in the billing data. This column MUST be of type String and SHOULD NOT contain null values. If a provider doesn't support a sub account construct (only has a billing account), use the BillingAccountName value. If a provider does support a sub account construct, but the charge does not apply to a sub account, the SubAccountName column may be null.
```

File: specification/dimensions/subaccountname.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+Work item: [Issues #18](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/18)
+
+Owner: John Grubb / Michael Arkoosh
+
+Spec Content
+------------
+
```
Comment (ijurica) [inline specification/dimensions/subaccountname.md:7]: a few notes:

- selected block should be removed
- Sub Account Name should be Heading L1 and all others Heading L2

File: supporting_content/dimensions/subaccountname.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Work item: [Issues #18](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/18)
+
+Owner: John Grubb / Michael Arkoosh
+
+Supporting content
+------------------
+
+#### Example provider mappings
```
[the remark this answers is not in the document]
Comment (rileyjenk) [reply supporting_content/dimensions/subaccountname.md:8]: After pulling some examples and looking various options for account structures in AWS here is what was found. 
1.) There are multiple ways to have multiple invoice generating accounts in AWS regardless if the are stand-alone, master-payer, or linked accounts.  This means that from an AWS side an account can be a sub account (linked account) to a higher order account (master payer account) and generate its own invoice and will not be included on the master payer.

That being said as far as FOCUS is concerned , any account that generates an invoice is considered a billing account. The provider may also specify it as a sub account. In the data there won't be a link in the data for true relationship.

**Pull request merged**